pkg/util/util.go

```diff
@@ -129,3 +133,85 @@ func Int32PtrDerefOr(ptr *int32, def int32) int32 {
 	}
 	return def
 }
+
+// RemoveAllOneFilesystem removes path and any children it contains.
+// It removes everything it can but returns the first error
+// it encounters. If the path does not exist, RemoveAll
+// returns nil (no error).
+// It makes sure it does not cross mount boundary, i.e. it does *not* remove
+// files from another filesystems. Like 'rm -rf --one-file-system'.
+// It is copied from RemoveAll() sources, with IsLikelyNotMountPoint
+func RemoveAllOneFilesystem(mounter mount.Interface, path string) error {
+	// Simple case: if Remove works, we're done.
+	err := os.Remove(path)
+	if err == nil || os.IsNotExist(err) {
+		return nil
+	}
+
+	// Otherwise, is this a directory we need to recurse into?
+	dir, serr := os.Lstat(path)
+	if serr != nil {
+		if serr, ok := serr.(*os.PathError); ok && (os.IsNotExist(serr.Err) || serr.Err == syscall.ENOTDIR) {
+			return nil
+		}
+		return serr
+	}
+	if !dir.IsDir() {
+		// Not a directory; return the error from Remove.
+		return err
+	}
+
+	// Directory.
+	isNotMount, err := mounter.IsLikelyNotMountPoint(path)
+	if err != nil {
+		return err
+	}
+	if !isNotMount {
+		return fmt.Errorf("cannot delete directory %s: it is a mount point", path)
+	}
+
+	fd, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			// Race. It was deleted between the Lstat and Open.
+			// Return nil per RemoveAll's docs.
+			return nil
+		}
+		return err
+	}
+
+	// Remove contents & return first error.
+	err = nil
+	for {
+		names, err1 := fd.Readdirnames(100)
```

Comment:
@wongma7 that's a question for [os.RemoveAll](https://golang.org/src/os/path.go)